### PR TITLE
feat(MediaLibrary) : add MediaLibrary and its usage within the SideMenu

### PR DESCRIPTION
**Pull request by FlorentMr**
## Describe your changes
add MediaLibrary and its usage within the SideMenu to use the workspace and its functionnalities

## Checklist tests
Utiliser le sidemenu pour charger des fichiers depuis MediaLibrary

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/providers/BoardProvider/types.ts [hunk 1]
```diff
@@ -19,6 +26,20 @@ export type BoardContextType = {
   isLoading: boolean;
   boardRights: Record<RightRole, boolean> | null;
   hasEditRights: () => boolean;
+  mediaLibraryRef: RefObject<MediaLibraryRef>;
+  libraryMedia: any;
```
Comment (niko-amosse) [inline frontend/src/providers/BoardProvider/types.ts:30]: pas de type ?

File: frontend/src/components/create-magnet/CreateMagnet.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  const { appCode } = useOdeClient();
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { mediaLibraryRef, libraryMedia, mediaLibraryHandlers, media } =
+    useMediaLibrary();
+
+  useEffect(() => {
+    if (libraryMedia) setIsOpen(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Mattox0) [inline frontend/src/components/create-magnet/CreateMagnet.tsx:17]: Ajoute ceci dans ton .eslintrc.cjs pour ne plus avoir besoin de le mettre

`"rules": {
"react-hooks/exhaustive-deps": "off"
}`

**Review by niko-amosse — APPROVED** [review]

**Review by Mattox0 — APPROVED** [review]

File: frontend/src/components/create-magnet/CreateMagnet.tsx (hunk 2 above)
Comment (alicedraillard) [reply frontend/src/components/create-magnet/CreateMagnet.tsx:17]: on l'a déjà ;) du cp faut juste enlever la ligne

File: frontend/src/components/create-magnet/style.ts
Comment (alicedraillard) [inline frontend/src/components/create-magnet/style.ts:1]: tu peux supprimer le fichier

File: frontend/src/components/board-view/BoardView.tsx [hunk 3]
```diff
@@ -12,13 +12,23 @@ import { useHeaderHeight } from "./useHeaderHeight";
...
 import { CardsVerticalLayout } from "../cards-vertical-layout/CardsVerticalLayout";
+import { CreateMagnet } from "../create-magnet/CreateMagnet";
 import { HeaderView } from "../header-view/HeaderView";
 import { SideMenu } from "../side-menu/SideMenu";
 import { ZoomComponent } from "../zoom-component/ZoomComponent";
 import { LAYOUT_TYPE } from "~/core/enums/layout-type.enum";
 import { useSideMenuData } from "~/hooks/useSideMenuData";
 import { useBoard } from "~/providers/BoardProvider";
 
+export interface MediaProps {
```
Comment (alicedraillard) [inline frontend/src/components/board-view/BoardView.tsx:23]: déjà créé un type pour, possible de l'appeler ?

File: frontend/src/providers/MediaLibraryProvider/index.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            ? `/workspace/document/${medialIb?._id}`
+            : (libraryMedia as string),
+        });
+      }
+    }
+  };
+
+  useEffect(() => {
+    updateLibraryMedia();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (alicedraillard) [inline frontend/src/providers/MediaLibraryProvider/index.tsx:93]: idem à enlever

File: frontend/src/providers/BoardProvider/types.ts (hunk 1 above)
Comment (ThomasSevagen) [reply frontend/src/providers/BoardProvider/types.ts:30]: en général améliorer le typage. tu as des types ultra ouverts comme les records

**Review by ThomasSevagen — APPROVED** [review]

**Review by alicedraillard — APPROVED** [review]

**Pull request merged**